### PR TITLE
fix(config): build all-langs.js before server

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

An error comes about in the server, if it is developed before the `all-langs.js` file is built. This PR resolves that.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "noEmit": true,
     "resolveJsonModule": true,
     "skipLibCheck": true,
-    "types": ["node", "jest", "@testing-library/jest-dom"]
+    "types": ["node", "jest", "@testing-library/jest-dom"],
+    "incremental": true
```
Comment (ojeytonwilliams) [inline tsconfig.json:26]:
```suggestion
    "types": ["node", "jest", "@testing-library/jest-dom"]
```
It doesn't seem to matter: if you pass `--incremental` it works regardless.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hey @ShaunSHamilton for the script to work it needs to be employed thusly:
```json
    "ensure-env": "cd ./tools/scripts/build/ && tsc --incremental && cross-env DEBUG=fcc:* node ./ensure-env.js",
```

Also, the tsconfig.buildinfo file this generates should be gitignored.

**Comment by nhcarrigan** [comment]
So I did a `git clean` to get a fresh start, and this is what I get if I run `develop:server`:

image

**Comment by ShaunSHamilton** [comment]
> So I did a `git clean` to get a fresh start, and this is what I get if I run `develop:server`:
> 
> image

I cannot reproduce. What steps did you exactly take? I tried:
1) `git clean -f`
2) `npm run clean`
3) `npm ci`
4) `npm run develop:server`

**Comment by ojeytonwilliams** [comment]
Thinking about it, could we remove the buildinfo as part of clean? Should help people get out of weird states. Removing/changing js without modifying ts files, for instance.

**Comment by nhcarrigan** [comment]
> What steps did you exactly take?

1. `git clean -idfx`
2. Kept `.env`, `**/node_modules/`, and `.husky`
3. `npm run develop:server`

**Comment by ShaunSHamilton** [comment]
> Thinking about it, could we remove the buildinfo as part of clean? Should help people get out of weird states. Removing/changing js without modifying ts files, for instance.

I do not understand the _weird states_ from it, but sure 👍

**Comment by ojeytonwilliams** [comment]
> I do not understand the weird states from it, but sure 

The TS and JS could get out of sync if you modified the JS. This shouldn't happen but it would be weird and `tsc` would not fix it. Not if you were building incrementally.

File: package.json [hunk 2]
```diff
@@ -29,8 +29,9 @@
     "build:client": "cd ./client && npm run build",
     "build:curriculum": "cd ./curriculum && npm run build",
     "build:server": "cd ./api-server && npm run build",
-    "clean": "npm-run-all clean:client clean:server clean:packages clean:root-deps",
+    "clean": "npm-run-all clean:build-files clean:client clean:server clean:packages clean:root-deps",
     "clean-and-develop": "npm run clean && npm ci && npm run develop",
+    "clean:build-files": "shx rm ./tools/scripts/build/tsconfig.tsbuildinfo",
```
Comment (ShaunSHamilton) [inline package.json:34]: @ojeytonwilliams Is this what you had in mind? Should I instead write a script for this, and include all the build files in it?

Comment (ojeytonwilliams) [reply package.json:34]: This looks good, I don't think we need anything more sophisticated.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by raisedadead — APPROVED** [review]

**Pull request merged**